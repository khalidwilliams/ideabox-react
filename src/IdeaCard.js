import React from 'react';

const IdeaCard = ({id, title, description, deleteCard, isFavorite, toggleFavorite}) => {
  let favoriteClass;
  isFavorite ? favoriteClass = 'favorited' : favoriteClass = '';

  return (
    <section id={id} className={favoriteClass}>
      <h3>{title}</h3>
      <p>{description}</p>
      <div>
        <button className="favorite-btn" onClick={() => toggleFavorite(id)}>Favorite</button>
        <button className="delete-btn" onClick={() => deleteCard(id)}>Delete</button>
      </div>
    </section>
  );

};

export default IdeaCard;
