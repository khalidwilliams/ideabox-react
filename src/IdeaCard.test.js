import React from 'react';
import { shallow } from 'enzyme';
import IdeaCard from './IdeaCard.js';

describe('Idea Card Component', () => {
  it('should render' , () => {
    const idea = shallow(<Idea/>)
    expect(idea).toBeDefined();
  });

  // describe('Idea Card Snapshots', () => {
  //   it('should match previous snapshot', () => {
  //     const idea = shallow(<Idea
  //         id={1}
  //         title="Dogs should vote"
  //         description="Give them rights"
  //         deleteCard={deleteCardMock}
  //       />);
  //
  //   });
  // });

  it('should render with a title', () => {
    const idea = shallow(<Idea id={1} title="Dogs should vote" description="Give them rights" />)
    const ideaTitle = <h3>Dogs should vote</h3>;

    expect(idea.contains(ideaTitle)).toEqual(true);
  });

  it('should render with a description', () => {
    const idea = shallow(<Idea id={1} title="Dogs should vote" description="Give them rights" />)
    const ideaBody = <p>Give them rights</p>;

    expect(idea.contains(ideaBody)).toEqual(true);
  })

  it('should not be favorited by default', () => {
    const idea = shallow(<Idea id={1} title="Dogs should vote" description="Give them rights" />)

    expect(idea).hasClass('favorited').toEqual(false);
  })

  it('should be able to be instantiated with a favorite', () => {
    const idea = shallow(<Idea id={1} title="Dogs should vote" description="Give them rights" isFavorite={true} />)

    expect(idea).hasClass('favorited').toEqual(true);
  });

  // it('should be able to be favorited', () => {
  //   const toggleFavoriteMock = jest.fn();
  //   const idea = shallow(<Idea
  //       id={1}
  //       title="Dogs should vote"
  //       description="Give them rights"
  //       toggleFavorite={toggleFavoriteMock}
  //     />);
  //
  //   expect(idea).hasClass('favorited').toEqual(false);
  //
  //   idea.find('favorite-btn').simulate('click');
  //
  //   expect(idea).hasClass('favorited').toEqual(true);
  //
  //   idea.find('favorite-btn').simulate('click');
  //
  //   expect(idea).hasClass('favorited').toEqual(true);
  // });

  it('should be able to be deleted', () => {
    const deleteCardMock = jest.fn();
    const idea = shallow(<Idea
        id={1}
        title="Dogs should vote"
        description="Give them rights"
        deleteCard={deleteCardMock}
      />);

    idea.find('.delete-btn').simulate('click');

    expect(deleteCardMock).toHaveBeenCalledWith(1);
  });
})
