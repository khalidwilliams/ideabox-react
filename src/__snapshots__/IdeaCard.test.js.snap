// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard Card Component IdeaCard Card Snapshots should render correctly based on required properties  1`] = `
<section
  className=""
  id={1}
>
  <h3>
    Dogs should vote
  </h3>
  <p>
    Give them rights
  </p>
  <div>
    <button
      className="favorite-btn"
      onClick={[Function]}
    >
      Favorite
    </button>
    <button
      className="delete-btn"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</section>
`;
